### PR TITLE
Ports Components from TG: "I don't actually know what I'm doing" edition

### PR DESCRIPTION
**Pull request by NonQueueingMatt**
title, basically

todo:

- [x] do the shit ninjanomnom told myazaki to do in the other pr

- [x] guide on components maybe?

### CODE REVIEW

**Comment by NonQueueingMatt** [comment]
!review

File: code/datums/components/multitool/multitool.dm [hunk 1]
```diff
@@ -66,7 +57,7 @@
 		M.set_buffer(null)
 		. = MT_REFRESH
 	else
-		. = on_topic(href, href_list, user)
+		. = on_topic_interaction.Invoke(href, href_list, user)
```
Comment (skull132) [inline code/datums/components/multitool/multitool.dm:60]: This should pass `src` as well.

File: html/changelogs/adds_components.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+author: mikomyazaki
```
Comment (skull132) [inline html/changelogs/adds_components.yml:1]:
```suggestion
author: mikomyazaki, MattAtlas
```

File: code/datums/components/README.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Datum Component System (DCS)
+
+## Concept
+
+Loosely adapted from /vg/. This is an entity component system for adding behaviours to datums when inheritance doesn't quite cut it. By using signals and events instead of direct inheritance, you can inject behaviours without hacky overloads. It requires a different method of thinking, but is not hard to use correctly. If a behaviour can have application across more than one thing. Make it generic, make it a component. Atom/mob/obj event? Give it a signal, and forward it's arguments with a `SendSignal()` call. Now every component that want's to can also know about this happening.
+
+See [this thread](https://tgstation13.org/phpBB/viewtopic.php?f=5&t=22674) for an introduction to the system as a whole.
```
Comment (Arrow768) [inline code/datums/components/README.md:7]: Would be good to have that somewhere in our wiki aswell.

**Pull request merged**